schema/crdb/README.adoc

```diff
@@ -56,41 +56,75 @@ See RFD 319 for more discussion of the online upgrade plans.
-  these `schema/crdb/NEW_VERSION/upN.sql` for as many `N` as you need, staring
-  with `N=1`.
-** Each file should contain _either_ one schema-modifying statement _or_ some
-   number of data-modifying statements.  You can combine multiple data-modifying
-   statements.  But you should not mix schema-modifying statements and
-   data-modifying statements in one file.  And you should not include multiple
-   schema-modifying statements in one file.
-** Beware that the entire file will be run in one transaction.  Expensive data-
-   modifying operations leading to long-running transactions are generally
-   to-be-avoided; however, there's no better way to do this today if you really
-   do need to update thousands of rows as part of the update.
-* Update `schema/crdb/dbinit.sql` to match what the database should look like
-  after your update is applied. Don't forget to update the version field of
-  `db_metadata` at the bottom of the file!
-** If necessary, do the same thing for `schema/crdb/dbwipe.sql`.
-* Update Nexus's idea of the latest schema, by updating its `SCHEMA_VERSION` to
-  `NEW_VERSION` within `nexus/db-model/src/schema.rs`.
-
-SQL Validation, via Automated Tests:
-
-* The `SCHEMA_VERSION` matches the version used in `dbinit.sql`
-* The combination of all `up.sql` files results in the same schema as
-  `dbinit.sql`
-* All `up.sql` files can be applied twice without error
+* Create a directory of SQL files describing how to update a database running
+  version `OLD_VERSION` to one running version `NEW_VERSION`:
+** Choose a unique, descriptive name for the directory.  It's suggested that
+   you stick to lowercase letters, numbers, and hyphen.  For example, if you're
+   adding a table called `widgets`, you might create a directory called
+   `create-table-widgets`.
+** Create the directory: `schema/crdb/NAME`.
+*** If only one SQL statement is necessary to get from `OLD_VERSION` to
+    `NEW_VERSION`, put that statement into `schema/crdb/NAME/up.sql`.  If
+    multiple statements are required, put each one into a separate file, naming
+    these `schema/crdb/NAME/upN.sql` for as many `N` as you need, staring with
+    `N=1`.
+*** Each file should contain _either_ one schema-modifying statement _or_ some
+    number of data-modifying statements.  You can combine multiple data-modifying
+    statements.  But you should not mix schema-modifying statements and
+    data-modifying statements in one file.  And you should not include multiple
+    schema-modifying statements in one file.
+*** Beware that the entire file will be run in one transaction.  Expensive data-
+    modifying operations leading to long-running transactions are generally
+    to-be-avoided; however, there's no better way to do this today if you really
+    do need to update thousands of rows as part of the update.
+* Update `schema/crdb/dbinit.sql`:
+** Update the SQL statements to match what the database should look like
+   after your up*.sql files are applied.
```

Comment:
Practically, I do this first, and save my schema changes for later. I wonder if we should recommend this first? "All tests except the schema tests" should pass if you only change `dbinit.sql`, and don't change the schema version.

Reply:
I think this is a good idea and I looked at it but it makes the instructions _more_ complicated because up front you update the schema parts of dbinit.sql and later you go back to it.  Reading it straight through it reads like "why did you tell me to go back to the same file twice"?  I'd welcome edits here but otherwise I'm just going to leave it.

I also think this will be less necessary now.  I did the same as you because I was worried about dealing with merges, but now that's pretty easy so I will probably just go ahead and do the migrations earlier than I used to.